env/io_posix.cc

```diff
@@ -56,6 +56,65 @@ int Fadvise(int fd, off_t offset, size_t len, int advice) {
 #endif
 }
 
+// On MacOS (and probably *BSD), the posix write and pwrite calls do not support
+// buffers larger than 2^31-1 bytes. These two wrappers fix this issue by
+// cutting the buffer in 1GB chunks when having to write buffers larger than
+// 2GB. We use this chunk size to be sure to keep the writes aligned.
+
+bool PosixWriteWrapper(int fd, const void* buf, size_t nbyte) {
+  constexpr size_t LIMIT_2GB = 1UL << 31;
+  constexpr size_t LIMIT_1GB = 1UL << 30;
```

Comment:
Why constexpr, rather than const?

I don't think we need to be so much fine granularity. We can always cut write limit to 1GB, to be simplified.

Also, the const naming convention is `kLimit1Gb`. See https://google.github.io/styleguide/cppguide.html#Constant_Names . And I personally prefer `kLimitOneGb` for this case.

Reply:
It is a habit for me to make compile-time constants `constexpr` and not `const`. 
My blame for the naming convention ...

Reply:
Nothing wrong with using `constexpr` here. I just feel that since we tend to use `const` in case like this in this code base, it might be a good idea to keep it inconsistency. I don't see a strong reason why we have to force a compile-time constant here. If you feel strongly about `constexpr`, you can keep it here.